api/v1/logic/cms/contests.js

```diff
@@ -0,0 +1,26 @@
+const CMS_SERVER = process.env.CMS_SERVER;
+const { get } = require('../../utils/fetch');
+const { ERRORS } = require("../../constants");
+const { CMS_ERRORS } = require("./errors");
+const { LogicError } = require("../../utils/errors");
+const { fetchWithToken } = require('./utils');
+
+async function getContest(contest_name) {
+  const url = `${CMS_SERVER}/api/contests/?name=${contest_name}`;
+  const body = await fetchWithToken({ method: get, url: url });
+  if (body.error != 0) {
+    if (body.error == CMS_ERRORS.NOT_FOUND.code) {
+      throw new LogicError(ERRORS.CMS_CONTEST_NOT_FOUND);
+    }
+    throw new LogicError(ERRORS.CMS_SERVER_ERROR);
+  }
```

Comment:
should flatten the condition either

```suggestion
  if (body.error && body.error ===  CMS_ERRORS.NOT_FOUND.code) {
    throw new LogicError(ERRORS.CMS_CONTEST_NOT_FOUND);
  } else if (body.error) {
    throw new LogicError(ERRORS.CMS_SERVER_ERROR);
  }
```
or 

```suggestion
  if (body.error) {
      throw new LogicError(body.error == CMS_ERRORS.NOT_FOUND.code                   
                         ? ERRORS.CMS_CONTEST_NOT_FOUND
                         : ERRORS.CMS_SERVER_ERROR);
  }
```

Reply:
Em cũng k hiểu em đã nghĩ gì khi code đống bùi nhùi này 
It should be 
```
if (body.error === CMS_ERRORS.NOT_FOUND.code) {
    throw new LogicError(ERRORS.CMS_CONTEST_NOT_FOUND);
  }
  if (body.error) {
    throw new LogicError({ ...ERRORS.CMS_FETCH_ERROR, data: { body } });
  }

```